backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQHealthCheck.java

```diff
@@ -46,27 +49,44 @@ public ComponentName componentName() {
     }
 
     @Override
-    public Result check() {
+    public Mono<Result> checkReactive() {
         try {
-            if (connectionPool.tryConnection() && rabbitChannelPoolImpl.tryChannel()) {
-                Optional<RabbitMQServerVersion> version = connectionPool.version();
+            return Flux.concat(connectionPool.tryConnection(),
+                rabbitChannelPoolImpl.tryChannel())
+                .reduce(true, (a, b) -> a && b)
```

Comment:
what about `.filter(status -> status == true).then(checkVersion()).switchIfEmpty(unhealthy)`?

Reply:
I prefer the reduce version.

Reply:
I know that you do, still I think it's overkill.
But do as you want.